Add a style guide and chapter template (#367)
theislab/single-cell-best-practices: CONTRIBUTING.md

# Contributing

## Our philosophy

> We aim for this book to be a major resource for introducing people to the field of single-cell sequencing data analysis and ensuring that newcomers and experienced analysis do things right.

Reviewer: ```suggestion
We aim for this book to be a resource for introducing people to single-cell and spatial data analysis while ensuring that newcomers and experienced analysts alike do things properly.
```

To ensure our recommendations reflect best practices, we try to rely exclusively on external and independent benchmarks.

## How can I contribute?

We warmly welcome contributions, especially those that help us keep our best practices up to date!
However, as mentioned [above](#our-philosophy), these recommendations are held to high standards.

If you're proposing new tutorials, corrections, or extensions, we strongly recommend opening an issue first to discuss your ideas with us.

> We're always eager to learn about the latest developments and will likely have follow-up questions — but we're genuinely excited to collaborate.

Reviewer: ```suggestion
We're always eager to learn about the latest developments and are happy to discuss the current state of best practices.
```

Author: Adopted!

> We also encourage contributions in the form of [quizzes and flashcards](#create-custom-quizzes-and-flashcards).

Reviewer: I get it but I'm not sure whether we want to add like 20 more questions per chapter. It'd be a lot to maintain and potentially verbose.

Author: I understand we should discuss this. I just took the content from the old `CONTRIBUTING.md` and updated it. What is the approximate/maximum number of questions we want per chapter? 5-15?

Reviewer 2: I think less than 5 would look good, but for bigger chapters 5-8 sounds good!

> Our collection of self-assessment questions is always growing, and community support in expanding them is greatly appreciated.

Reviewer: ```suggestion
```

Author: Adopted!

> The same applies to these contributions: Just open an issue and let's get in touch!

Reviewer: ```suggestion
The same applies to these contributions: Just open an issue and let's get in touch!
```

I'd rephrase this to a general request for issues to discuss proposals.

[![Open an Issue](https://img.shields.io/badge/Open%20Issue-blue?logo=github)](https://github.com/theislab/single-cell-best-practices/issues/new?title=Your+Issue+Title&body=Describe+your+issue+here)

## Book architecture

> The structure of the book is organized within the `jupyter_book` folder.

Reviewer: Weird sentence. I'd suggest that we rephrase this to the source of the book being in that folder or so

│   ├── ...
```

### Notebook Structure

Each `.ipynb` notebook should follow this standard structure:

1. 🧠 **Title**
2. 🔽 **Dropdown Section**
   - Key Takeaways
   - Environment Setup
   - Lamin Setup
3. 📖 **Main Content**
4. 🔗 **See Also** _(Dropdown)_
5. ❓ **Quiz / Flashcards**
6. 📚 **References**
7. 👥 **Contributors**

All dropdowns immediately following the title are automatically inserted if they meet the corresponding [requirements](#key-takeaways-environment-and-lamin-dropdown).
In addition, each chapter should conclude with a few questions that reinforce the main concepts covered.
[Below](#create-custom-quizzes-and-flashcards) you will find instructions on how to create quizzes and flashcards in our custom format.
We’ve also prepared a chapter [template](/jupyter-book/template/template.ipynb) with which you can quickly and effectively create a chapter for our book.

### Helpful links

- [Jupyter Book documentation](https://jupyterbook.org/en/stable/intro.html)
- [MyST Markdown documentation](https://mystmd.org)
- [Sphinx documentation](https://www.sphinx-doc.org/en/master/)

### Rules

- Place each sentence on its own line to make reviewing easier.

> - Use dropdowns whenever possible so that the reader is not overwhelmed by the content.

Reviewer: ```suggestion
- Use dropdowns so that the reader is not overwhelmed by the content.
```

I'll weaken this a bit although I pushed for this.

Author: Adopted!
